plugins/spi/SPIOutputTest.cpp

```diff
@@ -421,3 +425,184 @@ void SPIOutputTest::testCombinedP9813Control() {
                 backend.GetData(1, &length));
   OLA_ASSERT_EQ(0u, backend.Writes(1));
 }
+
+/**
+ * Test DMX writes in the individual APA102 mode.
+ */
+void SPIOutputTest::testIndividualAPA102Control() {
+  // setup Backend
+  FakeSPIBackend backend(2);
+  SPIOutput::Options options(0, "Test SPI Device");
+  // setup pixel_count to 2 (enough to test all cases)
+  options.pixel_count = 2;
+  // setup SPIOutput
+  SPIOutput output(m_uid, &backend, options);
+  // set personality to 7= Individual APA102
+  output.SetPersonality(7);
+  
+  // simulate incoming dmx data with this buffer
+  DmxBuffer buffer;
+  // setup an pointer to the returned data (the fake SPI data stream)
+  unsigned int length = 0;
+  const uint8_t *data = NULL;
+
+  // test1
+  // setup some 'DMX' data
+  buffer.SetFromString("1, 10, 100");
+  // simulate incoming data
+  output.WriteDMX(buffer);
+  // get fake SPI data stream
+  data = backend.GetData(0, &length);
+  // this is the expected spi data stream:
+  const uint8_t EXPECTED0[] = { 0, 0, 0, 0,               // StartFrame
+                                0xff, 0x64, 0x0a, 0x01,   // first Pixel
+                                0xff, 0, 0, 0,            // second Pixel
+                                0, 0, 0, 0, 0, 0, 0, 0};  // EndFrame
+  // check for Equality
+  OLA_ASSERT_DATA_EQUALS(EXPECTED0, arraysize(EXPECTED0), data, length);
+  // check if the output writes are 1
+  OLA_ASSERT_EQ(1u, backend.Writes(0));
+
+  // test2
+  buffer.SetFromString("255,128,0,10,20,30");
+  output.WriteDMX(buffer);
+  data = backend.GetData(0, &length);
+  const uint8_t EXPECTED1[] = { 0, 0, 0, 0, 
+                                0xff, 0, 0x80, 0xff,
+                                0xff, 0x1e, 0x14, 0x0a, 
+                                0, 0, 0, 0, 0, 0, 0, 0};
+  OLA_ASSERT_DATA_EQUALS(EXPECTED1, arraysize(EXPECTED1), data, length);
+  OLA_ASSERT_EQ(2u, backend.Writes(0));
+  
+  // test3
+  buffer.SetFromString("34,56,78");
+  output.WriteDMX(buffer);
+  data = backend.GetData(0, &length);
+  const uint8_t EXPECTED2[] = { 0, 0, 0, 0, 
+                                0xff, 0x4e, 0x38, 0x22,
+                                0xff, 0, 0, 0, 
```

Comment:
I think this is the wrong output. Sending a shorter frame shouldn't change what the values of the pixels outside the frame are.